tx.go

```diff
@@ -123,6 +127,52 @@ func FromContext(ctx context.Context) (Tx, bool) {
 	return t, ok
 }
 
+type DoneFunc func() error
+
+// TraceReq traces an http request. It returns a new context with the transaction
+// included in it, and a func to be called when the request is finished.
+//
+// Usage:
+//
+//     ctx, done := TraceRequest(ctx, name, req)
+//     defer done()
+func TraceRequest(ctx context.Context, name string, req *http.Request) (context.Context, DoneFunc, error) {
+	tx := NewRequestTx(name, req.URL.String())
+	ctx = WithTx(ctx, tx)
+	err := tx.Start()
+	df := func() error {
+		return tx.End()
+	}
+
+	return ctx, df, err
```

Comment:
I was actually thinking these top level functions need only return `DoneFunc`s and we just wouldn't call the "end" functions if start failed. That way the callers have a simpler api. They don't have to check the error twice (not like they can do much if start failed anyway, they're not going to abort their actual processing).

Something like:
```go
func TraceRequest(...) (.., DoneFunc) {
   err := tx.Start()
   df := func() error {
      if err {
         return err
      } else {
         return tx.End()
      }
   }
}
```